windows/ReactNativeCameraCPP/CameraRotationHelper.h

```diff
@@ -0,0 +1,61 @@
+﻿#pragma once
+#include "winrt/Windows.Graphics.Display.h"
+#include "CameraRotationHelper.g.h"
+
+// This is a Cpp/WinRT implementation of the Camera Rotation Helper class(C#) on MSDN:
+// https://docs.microsoft.com/en-us/windows/uwp/audio-video-camera/handle-device-orientation-with-mediacapture#camerarotationhelper-full-code-listing
+
+namespace winrt::ReactNativeCameraCPP::implementation
+{
+    struct CameraRotationHelper : CameraRotationHelperT<CameraRotationHelper>
+    {
+    public:
+        CameraRotationHelper(winrt::Windows::Devices::Enumeration::EnclosureLocation location);
+        winrt::event_token OrientationChanged(Windows::Foundation::EventHandler<bool> const& handler);
+        void OrientationChanged(winrt::event_token const& token) noexcept;
+
+        winrt::Windows::Devices::Sensors::SimpleOrientation GetCameraCaptureOrientation();
+        winrt::Windows::Storage::FileProperties::PhotoOrientation GetConvertedCameraCaptureOrientation();
+        winrt::Windows::Devices::Sensors::SimpleOrientation GetCameraPreviewOrientation();
+        int GetConvertedCameraPreviewOrientation();
+
+    private:
+
+        bool IsEnclosureLocationExternal(winrt::Windows::Devices::Enumeration::EnclosureLocation enclosureLocation)
+        {
+            return (enclosureLocation == nullptr || enclosureLocation.Panel() == winrt::Windows::Devices::Enumeration::Panel::Unknown);
+        }
+        bool ShouldMirrorPreview();
+        winrt::Windows::Devices::Sensors::SimpleOrientation GetCameraOrientationRelativeToNativeOrientation();
+
+        winrt::Windows::Devices::Sensors::SimpleOrientation SubtractOrientations(
+            winrt::Windows::Devices::Sensors::SimpleOrientation a,
+            winrt::Windows::Devices::Sensors::SimpleOrientation b);
+        winrt::Windows::Devices::Sensors::SimpleOrientation MirrorOrientation(winrt::Windows::Devices::Sensors::SimpleOrientation orientation);
+        winrt::Windows::Devices::Sensors::SimpleOrientation AddOrientations(winrt::Windows::Devices::Sensors::SimpleOrientation a, winrt::Windows::Devices::Sensors::SimpleOrientation b);
+
+        winrt::Windows::Storage::FileProperties::PhotoOrientation ConvertSimpleOrientationToPhotoOrientation(
+            winrt::Windows::Devices::Sensors::SimpleOrientation orientation);
+        int ConvertSimpleOrientationToClockwiseDegrees(winrt::Windows::Devices::Sensors::SimpleOrientation orientation);
```

Comment:
>ConvertSimpleOrientationToClockwiseDegrees [](start = 12, length = 42)

this should probably be a static as it doesn't access any of the members in the class

Reply:
will do